components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/SAMLSSOArtifactResolver.java

```diff
@@ -240,4 +245,38 @@ private ArtifactResponse buildArtifactResponse(Response response, ArtifactResolv
 
         return artifactResponse;
     }
+
+    private void startTenantFlow(String tenantDomain) throws IdentityException {
+
+        if (tenantDomain == null) {
+            return;
+        }
+
+        int tenantId;
+        try {
+            tenantId = SAMLSSOUtil.getRealmService().getTenantManager().getTenantId(tenantDomain);
+            if (tenantId == -1) {
+                // invalid tenantId, hence throw exception to avoid setting invalid tenant info.
+                String message = "Invalid Tenant Domain : " + tenantDomain;
+                if (log.isDebugEnabled()) {
+                    log.debug(message);
+                }
+                throw IdentityException.error(message);
+            }
+        } catch (UserStoreException e) {
+            String message = "Error occurred while getting tenant ID from tenantDomain " + tenantDomain;
+            log.error(message, e);
+            throw IdentityException.error(message, e);
```

Comment:
why are we logging and throwing the exception? If we do like that there is a chance of getting the stack trace printed two times in the console